#! /bin/sh /usr/share/dpatch/dpatch-run
## 04_fix_last-command-char.dpatch by Arnaud Fontaine <arnau@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: last-command-char has been removed in Emacs 24.3

@DPATCH@
--- elscreen-1.4.6~/elscreen.el	2013-04-15 09:29:42.722789497 +0900
+++ elscreen-1.4.6~/elscreen.el	2013-04-15 09:30:31.483311940 +0900
@@ -306,6 +306,11 @@
 
 ;;; Internal Functions:
 
+(defun elscreen-last-command-char ()
+  (if (featurep 'xemacs)
+      (event-to-character last-command-event)
+    last-command-event))
+
 (defvar elscreen-frame-confs nil
   "Alist that contains the information about screen configurations.")
 
@@ -989,7 +994,7 @@
 (defun elscreen-jump ()
   "Switch to specified screen."
   (interactive)
-  (let ((next-screen (string-to-number (string last-command-char))))
+  (let ((next-screen (string-to-number (string (elscreen-last-command-char)))))
     (if (and (<= 0 next-screen) (<= next-screen 9))
         (elscreen-goto next-screen))))
 (defalias 'elscreen-jump-0 'elscreen-jump)
